socket-saver.user.js

```diff
@@ -0,0 +1,27 @@
+// ==UserScript==
```

Comment:
Why is this a new userscript? Can't we conjoin the two quite easily? Instead of raising an event, inject a script.

We also have to think about the non-userscript scenario, the bookmarklet. From personal experience it's yet to fail me, but given that the userscript *sometimes* fails, maybe the bookmarklet *sometimes* fails too.